var/spack/repos/builtin/packages/rocwmma/package.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Rocwmma(CMakePackage):
+    """AMD's C++ library for accelerating mixed precision matrix multiplication
+       and accumulation (MFMA) operations leveraging specialized GPU matrix cores.
+       rocWMMA provides a C++ API to facilitate breaking down matrix multiply-accumulate
+       problems into fragments and using them in block-wise operations that are
+       distributed in parallel across GPU wavefronts. The API is a header library
+       of GPU device code meaning that matrix core acceleration may be compiled directly
+       into your kernel device code. This can benefit from compiler optimization in the
+       generation of kernel assembly, and does not incur additional overhead costs of
+       linking to external runtime libraries or having to launch separate kernels."""
+
+    homepage = "https://github.com/ROCmSoftwarePlatform/rocWMMA"
+    git = "https://github.com/ROCmSoftwarePlatform/rocWMMA.git"
+    url      = "https://github.com/ROCmSoftwarePlatform/rocWMMA/archive/refs/tags/rocm-5.2.0.tar.gz"
+    tags     = ['rocm']
+
+    maintainers = ['srekolam']
+
+    version('5.2.0', sha256='257ccd1cf2bc1d8064e72e78d276ef7446b2cb7e2dec05ff8331bb44eff2b7cb')
+
+    amdgpu_targets = ROCmPackage.amdgpu_targets
+
+    variant('amdgpu_target', values=auto_or_any_combination_of(*amdgpu_targets))
+    variant('build_type', default='Release', values=("Release", "Debug", "RelWithDebInfo"), description='CMake build type')
+
+    # gfx908:xnack-;gfx90a:xnack-;gfx90a:xnack+
+    # are only targets currently supported for @5.2.0
+    # releases
+    conflicts('amdgpu_target=gfx906:xnack-', when='@5.2.0')
+    conflicts('amdgpu_target=gfx908', when='@5.2.0')
+    conflicts('amdgpu_target=gfx1010', when='@5.2.0')
+    conflicts('amdgpu_target=gfx1011', when='@5.2.0')
+    conflicts('amdgpu_target=gfx1012', when='@5.2.0')
+    conflicts('amdgpu_target=gfx1030', when='@5.2.0')
+    conflicts('amdgpu_target=gfx1031', when='@5.2.0')
+
+    depends_on('cmake@3.16.8:', type='build', when='@5.2.0:')
+    depends_on('cmake@3.5:', type='build')
+
+    depends_on('googletest@1.10.0:', type='test')
+
+    for ver in ['5.2.0']:
+        depends_on('rocm-cmake@%s:' % ver, type='build', when='@' + ver)
+        depends_on('llvm-amdgpu@' + ver,  type='build',  when='@' + ver)
+        depends_on('hip@' + ver,                         when='@' + ver)
+        depends_on('rocblas@' + ver,  type='build',  when='@' + ver)
```

Comment:
You should propagate the amdgpu_target to the rocblas dependency. The rocsolver package is a good example of how to do that.